packages/flutter/lib/src/material/feedback.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+/// Provides platform-specific acoustic and/or haptic feedback for certain
+/// actions.
+///
+/// For example, to play the Android-typically click sound when a button is
+/// tapped, call [forTap]. For the Android-specific vibration when long pressing
+/// an element, call [forLongPress]. Alternatively, you can also wrap your
+/// [onTap] or [onLongPress] callback in [wrapForTap] or [wrapForLongPress] to
+/// achive the same (see example code below).
+///
+/// Calling any of these methods is a no-op on iOS as actions on that platform
+/// typically don't provide haptic or acoustic feedback.
+///
+/// All methods in this class are usually called from within a [build] method
+/// as you have to provide a [BuildContext].
+///
+/// ## Sample code
+///
+/// To trigger platform-specific feedback before executing the actual callback:
+///
+/// ```dart
+/// @override
```

Comment:
Ideally each example would be something you can copy and paste into code and have work right away, so here we would usually include the StatefulWidget and State boilerplate, as in:
https://master-docs-flutter-io.firebaseapp.com/flutter/painting/TextSpan/recognizer.html